lib/time.py

```diff
@@ -23,3 +55,37 @@ def sleep(secs):
 
 def time():
   return float(Now().UnixNano()) / Second
+
+
+_strftime_map = {
+    'Y': '2006', 'y': '06', 'b': 'Jan', 'B': 'January', 'm': '01', 'd': '02',
+    'a': 'Mon', 'A': 'Monday',
+    'H': '15', 'I': '03', 'M': '04', 'S': '05', 'L': '.000',
+    'p': 'PM',
+    'Z':  'MST', #'z': '-0700',
+    '%': '%',
+    # TODO: Implement c, x, X, j, U, w, W
+}
+
+def strftime(layout, tt=None): #pylint: disable=missing-docstring
+  ret = ''
+  t = (Unix(int(mktime(tt)), 0) if tt else Now()).Local()
+  prev, n = 0, layout.find('%')
+  while 0 <= n <= len(layout) - 2:
```

Comment:
Looks like it is from next line, strContains, looks like we have one number off in lineno.